mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -236,6 +236,15 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
   private def getDistinctSplits(splits: Array[Double]): Array[Double] = {
     splits(0) = Double.NegativeInfinity
     splits(splits.length - 1) = Double.PositiveInfinity
+
+    // -0.0 and 0.0 has different hashCode and they may be remained as two distinct value in splits
+    // but 0.0 > -0.0 is False and it will break parameter check (require strictly increasing order)
+    // so here normalize all -0.0 and 0.0 to be 0.0
```

Comment:
* Explain the current `distinct` behavior, which might not always use hashCode to distinct. I think the hash code are different only after boxing.
* It might be good to file a upstream bug.

Reply:
Done.